tests/rados/test_bug_fixes.py

```diff
@@ -0,0 +1,299 @@
+            log.exception(e)
+            return 1
+        finally:
+            log.info(
+                "\n \n ************** Execution of finally block begins here *************** \n \n"
+            )
+            # log cluster health
+            rados_obj.log_cluster_health()
+
+        log.info(
+            "Verification of ceph config show and ceph config get output has been completed"
+        )
+        return 0
+
+    if config.get("slow-osd-heartbeat"):
+        doc = (
+            "\n# CEPH-"
+            "\n Verify slow osd heartbeat warning being generated when network delay > 1 sec"
+            "\n\t 1. Deploy a cluster with at least 2 OSD hosts"
+            "\n\t 2. Ensure Cluster health condition should be HEALTH_OK"
+            "\n\t 3. Choose an OSD host at random and induce network delay greater than 1 > sec"
+            "\n\t 4. Slow OSD heartbeat warning should be displayed for all the OSDs part of the OSD host"
+        )
+
+        log.info(doc)
+        log.info("Running test to verify slow OSD heartbeat warning")
+
+        try:
+            # check the default value of network threshold
+            mgr_dump = rados_obj.run_ceph_command(cmd="ceph mgr dump", client_exec=True)
+            active_mgr = mgr_dump["active_name"]
+            _cmd = f"ceph tell mgr.{active_mgr} dump_osd_network"
+            dump_osd_network_out = rados_obj.run_ceph_command(
+                cmd=_cmd, client_exec=True
+            )
+            threshold = int(dump_osd_network_out["threshold"])
+
+            log.info(f"Current threshold for Slow OSD network is: {threshold}")
+            if threshold != 1000:
+                log.error(
+                    "Default threshold value for network delay is not 1000, aborting the test"
+                )
+                raise Exception(
+                    "Default threshold value for network delay is not 1000, aborting the test"
+                )
+
+            # fetch list of OSD hosts
+            osd_hosts = rados_obj.get_osd_hosts()
+            rand_host = random.choice(osd_hosts)
+            log.info(f"Chosen OSD host to add network delay: {rand_host}")
+
+            _cmd = f"ceph osd ls-tree {rand_host}"
+            osd_list = rados_obj.run_ceph_command(cmd=_cmd, client_exec=True)
+            log.info(f"List of OSDs on the host: {osd_list}")
+
+            assert rados_obj.add_network_delay_on_host(
+                hostname=rand_host, delay="1100ms", set_delay=True
+            )
+
+            # smart wait for 120 secs to check Slow OSD heartbeat warning
+            timeout_time = datetime.datetime.now() + datetime.timedelta(seconds=120)
+            while datetime.datetime.now() < timeout_time:
+                health_detail, _ = cephadm.shell(args=["ceph health detail"])
+                log.info(f"Health warning: \n {health_detail}")
+                if "Slow OSD heartbeats on back" not in health_detail:
```

Comment:
please check the logic here. Some confusion with if/else on the warning generation.